pkg/promtail/targets/file/tailer.go

```diff
@@ -136,31 +136,38 @@ func (t *tailer) markPositionAndSize() error {
 	t.posAndSizeMtx.Lock()
 	defer t.posAndSizeMtx.Unlock()
 
-	pos, err := t.tail.Tell()
+	size, err := t.tail.Size()
 	if err != nil {
+		// If the file no longer exists, no need to save position information
+		if err == os.ErrNotExist {
+			level.Info(t.logger).Log("msg", "skipping update of position for a file which does not currently exist")
+			return nil
+		}
 		return err
 	}
-	readBytes.WithLabelValues(t.path).Set(float64(pos))
-	t.positions.Put(t.path, pos)
+	totalBytes.WithLabelValues(t.path).Set(float64(size))
 
-	size, err := t.tail.Size()
+	pos, err := t.tail.Tell()
 	if err != nil {
 		return err
 	}
-	totalBytes.WithLabelValues(t.path).Set(float64(size))
+	readBytes.WithLabelValues(t.path).Set(float64(pos))
+	t.positions.Put(t.path, pos)
 
 	return nil
 }
 
-func (t *tailer) stop(removed bool) {
```

Comment:
In a change I made recently, I added this bool to do the conditional position update below but that isn't necessary (and was also not implemented correctly), it can be removed, calling the markPositionAndSize is basically a NOOP now if the file does not exist